pinctrl/hw_pwm.go

```diff
@@ -184,13 +186,34 @@ func (pwm *pwmDevice) disable() error {
 // Only call this from public functions, to avoid double-wrapping the errors.
 func (pwm *pwmDevice) wrapError(err error) error {
 	if err != nil {
-		return errors.Join(err, fmt.Errorf("HW PWM chipPath %s, line %d", pwm.chipPath, pwm.line))
+		return errors.Join(err, fmt.Errorf("HW PWM chipPath %s, line %d", pwm.chipPath, pwm.pwmLine))
 	}
 	return nil
 }
 
+/*
+For all pins belonging to the same bank, pin data is stored contiguously and in 8 byte chunks.
+For a given pin, this method determines:
+
+ 1. which bank the pin belongs to
```

Comment:
so I think we cheat on the "bank" check, because we know we are in bank 0 we just check to see if the pin is less than some hardcoded number on a lower line. the way we use the pi5 results in us only using bank 0, and eventually we may need to change this to handle other banks.

`if !(1 <= gpioNumber && gpioNumber <= numGPIOPins)`